src/backend/distributed/planner/query_colocation_checker.c

```diff
@@ -433,7 +433,7 @@ CreateTargetEntryForColumn(Form_pg_attribute attributeTuple, Index rteIndex,
 				attributeTuple->atttypmod, attributeTuple->attcollation, 0);
 	TargetEntry *targetEntry =
 		makeTargetEntry((Expr *) targetColumn, resno,
-						strdup(attributeTuple->attname.data), false);
+						pstrdup(attributeTuple->attname.data), false);
```

Comment:
not covered in tests, I suppose that explains why it does not show up in valgrind.

Reply:
seems to be a codecov issue, if I add a warning it definitely shows up in check-multi as expected